src/libs/actions/Transaction.js

```diff
@@ -72,8 +72,9 @@ function addStop(transactionID) {
  * @param {String} transactionID
  * @param {String} index
  * @param {Object} waypoint
+ * @param {Boolean} shouldAddToRecentWaypoints
  */
-function saveWaypoint(transactionID, index, waypoint) {
+function saveWaypoint(transactionID, index, waypoint, shouldAddToRecentWaypoints = true) {
```

Comment:
Rather than using a parameter for this, the code should automatically filter out any waypoints with an address of "Your location" (this is what we do on the server to keep it from being saved).

Reply:
Yes, let's put that value in a constant.

For the case where someone enters "Your location", I think that would just be considered an invalid address and so it is fine to filter it out.